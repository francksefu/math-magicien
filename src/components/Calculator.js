import { useState } from 'react';
import '../App.css';
import calculate from '../logic/calculate';

function Calculator() {
  const [result, setResult] = useState({});
  const handleClick = (e) => {
    setResult(calculate(result, e.target.textContent));
  };

  return (
    <section>
      <h2>Calculator</h2>
      <p style={{ padding: '10px', marginTop: '3%', textAlign: 'center' }}>Let' do some math : </p>
      <div className="tableau">{result.next === null ? result.total : result.next}</div>
      <div className="ligne">
        <Span classons="aucun" valeur="AC" cliquons={handleClick} />
        <Span classons="aucun" valeur="+/-" cliquons={handleClick} />
        <Span classons="aucun" valeur="%" cliquons={handleClick} />
        <Span classons="orange" valeur="รท" cliquons={handleClick} />
      </div>
      <div className="ligne">
        <Span classons="aucun" valeur="7" cliquons={handleClick} />
        <Span classons="aucun" valeur="8" cliquons={handleClick} />
        <Span classons="aucun" valeur="9" cliquons={handleClick} />
        <Span classons="orange" valeur="x" cliquons={handleClick} />
      </div>
      <div className="ligne">
        <Span classons="aucun" valeur="4" cliquons={handleClick} />
        <Span classons="aucun" valeur="5" cliquons={handleClick} />
        <Span classons="aucun" valeur="6" cliquons={handleClick} />
        <Span classons="orange" valeur="-" cliquons={handleClick} />
      </div>
      <div className="ligne">
        <Span classons="aucun" valeur="1" cliquons={handleClick} />
        <Span classons="aucun" valeur="2" cliquons={handleClick} />
        <Span classons="aucun" valeur="3" cliquons={handleClick} />
        <Span classons="orange" valeur="+" cliquons={handleClick} />
      </div>
      <div className="ligne">
        <Span classons="deux" valeur="0" cliquons={handleClick} />
        <Span classons="dernier" valeur="." cliquons={handleClick} />
        <Span classons="dernier orange" valeur="=" cliquons={handleClick} />
      </div>
    </section>
  );
}

const Span = (props) => (
  <span className={props.classons} onClick={props.cliquons}>
    {props.valeur}
  </span>
);

export default Calculator;
