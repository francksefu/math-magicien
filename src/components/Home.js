function Home() {
  return (
    <div className="home">
      <h2>Home Welcome</h2>
      <p>
        Hello everyone, this is just a page, for trying
        Hello everyone, this is just a page, for trying
        Dans cette leçon, nous nous sommes concentrés principalement sur les
        options CSS dont vous disposez dans l’ARC. Si vous voulez en savoir
        plus sur le soutien CSS de l’ARC et sur la façon de le personnaliser, lisez
        la documentation de l’ARC, en particulier la section Styles et actifs.
      </p>
      <p>
        Dans cette leçon, nous nous sommes concentrés principalement sur
        les options CSS dont vous disposez dans l’ARC. Si vous voulez en
        savoir plus sur le soutien CSS de l’ARC et sur la façon de le personnaliser,
        lisez la documentation de l’ARC, en particulier la section Styles et actifs.
      </p>
    </div>

  );
}
export default Home;
