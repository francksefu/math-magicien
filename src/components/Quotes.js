import { useState, useEffect } from 'react';

function Quotes() {
  const [data, setData] = useState([]);
  const [hasError, setHasError] = useState(false);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    const send = () => {
      setLoading(true);
      try{
        fetch('https://api.api-ninjas.com/v1/quotes?category=money', {
          method: 'GET',
          headers: { 'X-Api-Key': 'tTySwiau1PsXJs5wvhf1dw==NK8bNzv70yBdLNYz' },
          contentType: 'application/json',
        })
          .then((response) => response.json())
          .then((json) => setData(json));
      } catch {
        setHasError(true)
      }
        setLoading(false)
      };
    send();
  }, [setData, setLoading]);

  if (hasError) return <h2>Something went wrong!</h2>;
  if (loading) return <h2> Loading ...</h2>;
  return (
    <div>
      { data.length ? (
      <div className="quotation">
        <h2>Quotes</h2>
        <p>{data[0].quote}</p>
        <p><small>
          author : {data[0].author}
        </small></p>
        <p><small>
          category : {data[0].category}
        </small></p>
      </div>
    ) : (<p>Loading..</p>)}
    </div>
  );
}

export default Quotes;
