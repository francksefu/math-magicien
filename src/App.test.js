import { render } from '@testing-library/react';
import Calculator from './components/Calculator';
import Home from './components/Home';
import Quotes from './components/Quotes';

// import TestRenderer from 'react-test-renderer';
test('renders learn react link', () => {
  expect(render(<Calculator />)).toMatchSnapshot();
  expect(render(<Home />)).toMatchSnapshot();
  expect(render(<Quotes />)).toMatchSnapshot();
});
