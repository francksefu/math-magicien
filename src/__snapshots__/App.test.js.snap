// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <h2>
          Calculator
        </h2>
        <p
          style="margin-top: 3%; text-align: center; padding: 10px;"
        >
          Let' do some math : 
        </p>
        <div
          class="tableau"
        />
        <div
          class="ligne"
        >
          <button
            class="aucun"
            type="button"
          >
            AC
          </button>
          <button
            class="aucun"
            type="button"
          >
            +/-
          </button>
          <button
            class="aucun"
            type="button"
          >
            %
          </button>
          <button
            class="orange"
            type="button"
          >
            รท
          </button>
        </div>
        <div
          class="ligne"
        >
          <button
            class="aucun"
            type="button"
          >
            7
          </button>
          <button
            class="aucun"
            type="button"
          >
            8
          </button>
          <button
            class="aucun"
            type="button"
          >
            9
          </button>
          <button
            class="orange"
            type="button"
          >
            x
          </button>
        </div>
        <div
          class="ligne"
        >
          <button
            class="aucun"
            type="button"
          >
            4
          </button>
          <button
            class="aucun"
            type="button"
          >
            5
          </button>
          <button
            class="aucun"
            type="button"
          >
            6
          </button>
          <button
            class="orange"
            type="button"
          >
            -
          </button>
        </div>
        <div
          class="ligne"
        >
          <button
            class="aucun"
            type="button"
          >
            1
          </button>
          <button
            class="aucun"
            type="button"
          >
            2
          </button>
          <button
            class="aucun"
            type="button"
          >
            3
          </button>
          <button
            class="orange"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="ligne"
        >
          <button
            class="deux"
            type="button"
          >
            0
          </button>
          <button
            class="dernier"
            type="button"
          >
            .
          </button>
          <button
            class="dernier orange"
            type="button"
          >
            =
          </button>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <h2>
        Calculator
      </h2>
      <p
        style="margin-top: 3%; text-align: center; padding: 10px;"
      >
        Let' do some math : 
      </p>
      <div
        class="tableau"
      />
      <div
        class="ligne"
      >
        <button
          class="aucun"
          type="button"
        >
          AC
        </button>
        <button
          class="aucun"
          type="button"
        >
          +/-
        </button>
        <button
          class="aucun"
          type="button"
        >
          %
        </button>
        <button
          class="orange"
          type="button"
        >
          รท
        </button>
      </div>
      <div
        class="ligne"
      >
        <button
          class="aucun"
          type="button"
        >
          7
        </button>
        <button
          class="aucun"
          type="button"
        >
          8
        </button>
        <button
          class="aucun"
          type="button"
        >
          9
        </button>
        <button
          class="orange"
          type="button"
        >
          x
        </button>
      </div>
      <div
        class="ligne"
      >
        <button
          class="aucun"
          type="button"
        >
          4
        </button>
        <button
          class="aucun"
          type="button"
        >
          5
        </button>
        <button
          class="aucun"
          type="button"
        >
          6
        </button>
        <button
          class="orange"
          type="button"
        >
          -
        </button>
      </div>
      <div
        class="ligne"
      >
        <button
          class="aucun"
          type="button"
        >
          1
        </button>
        <button
          class="aucun"
          type="button"
        >
          2
        </button>
        <button
          class="aucun"
          type="button"
        >
          3
        </button>
        <button
          class="orange"
          type="button"
        >
          +
        </button>
      </div>
      <div
        class="ligne"
      >
        <button
          class="deux"
          type="button"
        >
          0
        </button>
        <button
          class="dernier"
          type="button"
        >
          .
        </button>
        <button
          class="dernier orange"
          type="button"
        >
          =
        </button>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders learn react link 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <h2>
          Calculator
        </h2>
        <p
          style="margin-top: 3%; text-align: center; padding: 10px;"
        >
          Let' do some math : 
        </p>
        <div
          class="tableau"
        />
        <div
          class="ligne"
        >
          <button
            class="aucun"
            type="button"
          >
            AC
          </button>
          <button
            class="aucun"
            type="button"
          >
            +/-
          </button>
          <button
            class="aucun"
            type="button"
          >
            %
          </button>
          <button
            class="orange"
            type="button"
          >
            รท
          </button>
        </div>
        <div
          class="ligne"
        >
          <button
            class="aucun"
            type="button"
          >
            7
          </button>
          <button
            class="aucun"
            type="button"
          >
            8
          </button>
          <button
            class="aucun"
            type="button"
          >
            9
          </button>
          <button
            class="orange"
            type="button"
          >
            x
          </button>
        </div>
        <div
          class="ligne"
        >
          <button
            class="aucun"
            type="button"
          >
            4
          </button>
          <button
            class="aucun"
            type="button"
          >
            5
          </button>
          <button
            class="aucun"
            type="button"
          >
            6
          </button>
          <button
            class="orange"
            type="button"
          >
            -
          </button>
        </div>
        <div
          class="ligne"
        >
          <button
            class="aucun"
            type="button"
          >
            1
          </button>
          <button
            class="aucun"
            type="button"
          >
            2
          </button>
          <button
            class="aucun"
            type="button"
          >
            3
          </button>
          <button
            class="orange"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="ligne"
        >
          <button
            class="deux"
            type="button"
          >
            0
          </button>
          <button
            class="dernier"
            type="button"
          >
            .
          </button>
          <button
            class="dernier orange"
            type="button"
          >
            =
          </button>
        </div>
      </section>
    </div>
    <div>
      <div
        class="home"
      >
        <h2>
          Home Welcome
        </h2>
        <p>
          Hello everyone, this is just a page, for trying Hello everyone, this is just a page, for trying Dans cette leรงon, nous nous sommes concentrรฉs principalement sur les options CSS dont vous disposez dans lโARC. Si vous voulez en savoir plus sur le soutien CSS de lโARC et sur la faรงon de le personnaliser, lisez la documentation de lโARC, en particulier la section Styles et actifs.
        </p>
        <p>
          Dans cette leรงon, nous nous sommes concentrรฉs principalement sur les options CSS dont vous disposez dans lโARC. Si vous voulez en savoir plus sur le soutien CSS de lโARC et sur la faรงon de le personnaliser, lisez la documentation de lโARC, en particulier la section Styles et actifs.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home"
    >
      <h2>
        Home Welcome
      </h2>
      <p>
        Hello everyone, this is just a page, for trying Hello everyone, this is just a page, for trying Dans cette leรงon, nous nous sommes concentrรฉs principalement sur les options CSS dont vous disposez dans lโARC. Si vous voulez en savoir plus sur le soutien CSS de lโARC et sur la faรงon de le personnaliser, lisez la documentation de lโARC, en particulier la section Styles et actifs.
      </p>
      <p>
        Dans cette leรงon, nous nous sommes concentrรฉs principalement sur les options CSS dont vous disposez dans lโARC. Si vous voulez en savoir plus sur le soutien CSS de lโARC et sur la faรงon de le personnaliser, lisez la documentation de lโARC, en particulier la section Styles et actifs.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders learn react link 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <h2>
          Calculator
        </h2>
        <p
          style="margin-top: 3%; text-align: center; padding: 10px;"
        >
          Let' do some math : 
        </p>
        <div
          class="tableau"
        />
        <div
          class="ligne"
        >
          <button
            class="aucun"
            type="button"
          >
            AC
          </button>
          <button
            class="aucun"
            type="button"
          >
            +/-
          </button>
          <button
            class="aucun"
            type="button"
          >
            %
          </button>
          <button
            class="orange"
            type="button"
          >
            รท
          </button>
        </div>
        <div
          class="ligne"
        >
          <button
            class="aucun"
            type="button"
          >
            7
          </button>
          <button
            class="aucun"
            type="button"
          >
            8
          </button>
          <button
            class="aucun"
            type="button"
          >
            9
          </button>
          <button
            class="orange"
            type="button"
          >
            x
          </button>
        </div>
        <div
          class="ligne"
        >
          <button
            class="aucun"
            type="button"
          >
            4
          </button>
          <button
            class="aucun"
            type="button"
          >
            5
          </button>
          <button
            class="aucun"
            type="button"
          >
            6
          </button>
          <button
            class="orange"
            type="button"
          >
            -
          </button>
        </div>
        <div
          class="ligne"
        >
          <button
            class="aucun"
            type="button"
          >
            1
          </button>
          <button
            class="aucun"
            type="button"
          >
            2
          </button>
          <button
            class="aucun"
            type="button"
          >
            3
          </button>
          <button
            class="orange"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="ligne"
        >
          <button
            class="deux"
            type="button"
          >
            0
          </button>
          <button
            class="dernier"
            type="button"
          >
            .
          </button>
          <button
            class="dernier orange"
            type="button"
          >
            =
          </button>
        </div>
      </section>
    </div>
    <div>
      <div
        class="home"
      >
        <h2>
          Home Welcome
        </h2>
        <p>
          Hello everyone, this is just a page, for trying Hello everyone, this is just a page, for trying Dans cette leรงon, nous nous sommes concentrรฉs principalement sur les options CSS dont vous disposez dans lโARC. Si vous voulez en savoir plus sur le soutien CSS de lโARC et sur la faรงon de le personnaliser, lisez la documentation de lโARC, en particulier la section Styles et actifs.
        </p>
        <p>
          Dans cette leรงon, nous nous sommes concentrรฉs principalement sur les options CSS dont vous disposez dans lโARC. Si vous voulez en savoir plus sur le soutien CSS de lโARC et sur la faรงon de le personnaliser, lisez la documentation de lโARC, en particulier la section Styles et actifs.
        </p>
      </div>
    </div>
    <div>
      <div>
        <p>
          Loading..
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p>
        Loading..
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
